### DISCUSSION TITLE
Supply "Review and Publish" with Managed Publishing

### DISCUSSION THREAD

**Discussion by JaerenMoneyBox** [fastlane/fastlane#19673] [Q&A]
Hey,

We are trying to automate our full release cycle using fastlane + supply.  We use ["Managed Publishing"](https://support.google.com/googleplay/android-developer/answer/9859654?hl=en-GB) which means we "submit" our changes for review, then once they have been approved we can then "Review and Publish" which pushes the changes to our users.

![image](https://user-images.githubusercontent.com/16168351/144708305-91ecf12a-c4ff-40ab-9611-b0513d92c396.png)

We can automate the first step to submit the changes.  However we cannot work out how to automate the "Review and Publish" step.  We have tried running this code below, which targets the latest version and tries to publish that version.  It doesn't give an error but unfortunately has no effect.

```
    supply(
      version_code: last_git_tag,
      track: 'beta',
      track_promote_to: 'beta'
    )
```

Do we know if this is possible?  If not, would it be possible to add this functionality?

Thanks.

**Comment by Irina369** [comment]
Hello,

Our team would also like to have the possibility to automate the Managed Publishing workflow with Fastlane and do the following:
1. Set Managed Publishing status to ON/OFF
2. Release the update/changes in the '_Review and Publish_' step

Can you please suggest whether an issue should be opened for this or is this already being worked on?

Thank you!

**Comment by DouglasSouzaGomes** [comment]
Just adding a bit to this issue, I've done some tests trying to publish an apps' new version that had already been approved in the review, but has "Managed Publishing" on.

I figured the most relevant supply keys would be `track`, `track_promote_to`, `track_promote_release_status` and `release_status`, so I messed around with their values. What I think happens is, the app that's been approved on review is already in the "production" track (correct me if I'm wrong), it's just not on the "completed" release status. So trying to change the track to production won't really work. The lane runs fine, no errors, but nothing really happens to the app. 

However, trying to change the release status also doesn't work. I tried setting it to "completed", but, just like I mentioned above, the lane runs just fine, but nothing happens to the latest app version waiting to be published.

Sorry if this has already been addressed somewhere else, I couldn't find it.

**Comment by mataku** [comment]
We couldn't do that because Google Play Developer API does not currently provide an API to release changes via managed publishing. 

Related issue about track status: https://issuetracker.google.com/issues/179708468

**Reply by DouglasSouzaGomes** [reply]
Oh, I see. That's quite unfortunate, but thanks for the feedback!
